db-async-common/src/main/java/com/github/jasync/sql/db/pool/ActorBasedObjectPool.kt

```
@@ -496,6 +498,10 @@ private class ObjectPoolActor<T : PooledObject>(
         when (tried) {
             is Failure -> throw tried.exception
         }
+        val age = System.currentTimeMillis() - a.creationTime
+        if (configuration.maxTtl > 0 && age > configuration.maxTtl) {
+            throw ObjectAgedOutException(a, age, configuration.maxTtl)
```

I don't think we should throw an exception in case of ttl passed, just destroy the item on validation

---

Should validate() call dispose directly then?

---

Calling `item.destroy()` directly, let me know if that's not what you had in mind.

---

Looking at the code again it seems that you did the right thing. The only comment then is to call the exception MaxTtlPassedException

---

:+1: